src/init.jl

```diff
@@ -116,16 +125,13 @@ function __init__()
             return res, back(ones(Int, size(θ)))[1]
         end
 
-        # NOTE
-        # It's important for Zygote to come after ForwardDiff otherwise
-        # the function def below will be override by ForwardDiff's def.
-        # This is probably because Zygote is dependent on ForwardDiff and
-        # they will be also available at the same time.
-        function Hamiltonian(metric, ℓπ)
+        function Hamiltonian(metric::AbstractMetric, ℓπ, ::Union{ZygoteAD, Type{ZygoteAD}})
             ∂ℓπ∂θ(θ::AbstractVecOrMat) = ∂ℓπ∂θ_zygote(ℓπ, θ)
             return Hamiltonian(metric, ℓπ, ∂ℓπ∂θ)
         end
 
+        export ZygoteAD

```

Comment:
Same problem as above, ZygoteAD will be defined in Turing as well:
https://github.com/TuringLang/Turing.jl/blob/93f1ff5891185eace12ec529c22dc2367a1a7946/src/core/ad.jl#L35